plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextJavaValidator.java

```diff
@@ -409,78 +464,81 @@ public void checkValueReferenedBeforeDefined(Scope scope) {
 		EList<Declaration> declarations = scope.getDeclarations();
 		Set<QualifiedName> defined = Sets.newHashSet();
 		for (Declaration declaration : declarations) {
-			if (declaration instanceof VariableDefinition) {
-				VariableDefinition definition = (VariableDefinition) declaration;
-				if (!definition.isConst())
-					return;
-				Expression initialValue = definition.getInitialValue();
-				List<Expression> toCheck = Lists.newArrayList(initialValue);
-				TreeIterator<EObject> eAllContents = initialValue.eAllContents();
-				while (eAllContents.hasNext()) {
-					EObject next = eAllContents.next();
-					if (next instanceof Expression)
-						toCheck.add((Expression) next);
-				}
-				for (Expression expression : toCheck) {
-					EObject referencedObject = null;
-					if (expression instanceof FeatureCall)
-						referencedObject = ((FeatureCall) expression).getFeature();
-					else if (expression instanceof ElementReferenceExpression)
-						referencedObject = ((ElementReferenceExpression) expression).getReference();
-					if (referencedObject instanceof VariableDefinition) {
-						if (!defined.contains(nameProvider.getFullyQualifiedName(referencedObject)))
-							error(REFERENCE_CONSTANT_BEFORE_DEFINED, definition,
-									TypesPackage.Literals.PROPERTY__INITIAL_VALUE);
-					}
+			if (!(declaration instanceof VariableDefinition)) {
+				continue;
```

Comment:
We rarely use `continue` or `break` in SCT, so the old version was more readable, especially because it didn't use inverted logic. The old version basically said "if A do B", now you say "if not A do something unimportant, (implicit) else do something important".